### DISCUSSION TITLE
q-input is cleared when out of focus

### DISCUSSION THREAD

**Discussion by cold-rain-solutions** [quasarframework/quasar#11570] [General - Components / Directives / etc] [answered]
q-input value is cleared when out of focus. Props entry is set to Object type and default values are set.
So when you finish entering text it is cleared when you click out from input. If you try to use simple <input with v-model everything works. How to use q-input with Object props default value?

```
<div id="q-app">
    
  <div class="q-ma-md">
    <blog-post />
  </div>

</div>
```

```
const { useQuasar } = Quasar

const app = Vue.createApp({
  setup (props) {
    const $q = useQuasar()
    
    function notify () {
      $q.notify('Running on Quasar v' + $q.version)
    }
    
    return {
      notify
    }
  }
})

app.use(Quasar, { config: {} })
app.component('blogPost', {
  props: {
    entry: {
      type: Object,
      default() {
        return {
          nested: ''
        }
      }
    }
  },
  template: '<q-input v-model="entry.nested" label="nested" />'
})
app.mount('#q-app')

```

**Comment by pdanpdan** [comment, marked as answer]
you cannot modify props - props are for communication from parent to child
to change the external value you must emit and change the value in the parent

**Reply by cold-rain-solutions** [reply]
But I did not have any issues with vuetify and 
```
<input type="text v-model="entry.nested" /> 
```
Also if I pass 
```
<blog-post :entry="{neseted: ''}"/> 
```
then it works.

**Reply by hawkeye64** [reply]
Objects and Arrays are an exception because you are not changing the value (an array is an array, an object is still an object, but a string or number doesn't follow that paradigm), just the contents. But, if you care, this would be an anti-pattern that makes it in the future harder to debug, because something is happening in a way not expected.

**Reply by cold-rain-solutions** [reply]
Thanks for replay, I got it. I am using Laravel+InertiaJs so in response entry data is set as prop to Page component. When I used vuetify it was working without any issues bet when I change to q-input is stopped to work.  Thanks for the clarification.
